adapters/Mintegral/MintegralAdapter/GADMediationAdapterMintegral.m

```diff
@@ -4,44 +4,97 @@
 //
 // Unless required by applicable law or agreed to in writing, software
 // distributed under the License is distributed on an "AS IS" BASIS,
 // WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 // See the License for the specific language governing permissions and
 // limitations under the License.
-//
 
 #import "GADMediationAdapterMintegral.h"
-
 #import "GADMediationAdapterMintegralConstants.h"
+#import "GADMAdapterMintegralExtras.h"
+#import "GADMAdapterMintegralRewardedAdRenderer.h"
+#import "GADMAdapterMintegralBannerRenderer.h"
+#import "GADMAdapterMintegralInterstitialRenderer.h"
+#import "GADMAdapterMintegralNativeRenderer.h"
+#import "GADMAdapterMintegralUtils.h"
+#import <MTGSDK/MTGSDK.h>
+#import <MTGSDKBidding/MTGBiddingSDK.h>
+
+
+@implementation GADMediationAdapterMintegral {
+    //Mintegral Mediation rewarded ad.
+    GADMAdapterMintegralRewardedAdRenderer * _rewardedAd;
+    
+    //Mintegral Mediation banner ad.
+    GADMAdapterMintegralBannerRenderer *_banner;
+    
+    //Mintegral Mediation interstitial ad.
+    GADMAdapterMintegralInterstitialRenderer *_interstitialAd;
+    
+    //Mintegral Mediation native ad.
+    GADMAdapterMintegralNativeRenderer *_nativeAd;
+}
 
-@implementation GADMediationAdapterMintegral
 
-- (void)collectSignalsForRequestParameters:(nonnull GADRTBRequestParameters *)params
-                         completionHandler:
-                             (nonnull GADRTBSignalCompletionHandler)completionHandler {
-  // TODO: Start signal generation and call completionHandler when the signal generation finishes or
-  // fails.
-  completionHandler("", nil);
+#pragma mark GADRTBAdapter
+
++ (nullable Class<GADAdNetworkExtras>)networkExtrasClass {
+    return [GADMAdapterMintegralExtras class];
 }
 
-+ (void)setUpWithConfiguration:(nonnull GADMediationServerConfiguration *)configuration
-             completionHandler:(nonnull GADMediationAdapterSetUpCompletionBlock)completionHandler {
-  // TODO: Set up Mintegral SDK and perform any necessary prefetching or configuration work.
-  completionHandler(nil);
++ (void)setUpWithConfiguration:(GADMediationServerConfiguration *)configuration
+             completionHandler:(GADMediationAdapterSetUpCompletionBlock)completionHandler {
+    NSString *appKey  = @"";
+    NSString *appId = @"";
+    NSError *error = nil;
+    if ([configuration.credentials isKindOfClass:NSArray.class] &&
+        configuration.credentials.count > 0) {
+        appId = configuration.credentials[0].settings[GADMAdapterMintegralAppID];
+        appKey = configuration.credentials[0].settings[GADMAdapterMintegralAppKey];
+        if ([GADMAdapterMintegralUtils isEmpty:appId]) {
+            error = GADMAdapterMintegralErrorWithCodeAndDescription(GADMintegralErrorInvalidServerParameters,@"AppId is nil");
+            if (completionHandler) {
+                completionHandler(error);
+                return;
+            }
+        }
```

Comment:
Based on our kickoff call, app key and ID are for account-level and app-level used respectively, so we should add error handling when multiple app keys and IDs were found.

Consider this scenario: Under the same app in AdMob account, the publisher might (accidentally) use different Mintegral App IDs and/or keys for two different AdMob ad units.

Implementation example: https://github.com/googleads/googleads-mobile-ios-mediation/blob/master/adapters/Fyber/FyberAdapter/GADMediationAdapterFyber.m#L61-L85